package rpcframe;
message RpcInnerReq {
    required string service_name = 1;
    required string method_name = 2;
    required string request_id = 3;
    enum RPC_TYPE {
        ONE_WAY = 0;
        TWO_WAY = 1;
    }
    required RPC_TYPE type = 4;
    required uint32 timeout = 5;  //unit second
    required bytes data = 6;
}

message RpcInnerResp {
    required string request_id = 1; //should be the same with RpcInnerReq::request_id
    required uint32 ret_val = 2;    //the status returned from server side ref: common/include/RpcDefs.h
    required bytes data = 3;
}
